### ISSUE TITLE
Breaks on my test .mkv file

### ISSUE THREAD

**Issue by ddorian** [slhck/ffmpeg-bitrate-stats#13]
**Checklist** (please tick all boxes)
- [x] I am using the latest version of `ffmpeg-bitrate-stats` (run `pip3 install --upgrade ffmpeg-bitrate-stats`)
- [x] I am using the latest stable version of `ffmpeg` or a recent build from Git master

**Expected behavior**

I expected it to work using this video [out.zip](https://github.com/user-attachments/files/17599617/out.zip)

**Actual behavior**

See trace.

**Command**
The exact command you were trying to run:

```
ffmpeg-bitrate-stats out.mkv -v
```

Any output you get when running the command with the `-v` flag:

```python
 ffmpeg-bitrate-stats out.mkv -v
DEBUG - Calculating frame size from out.mkv
DEBUG - [cmd] ffprobe -loglevel error -select_streams v:0 -read_intervals % -show_packets -show_entries packet=pts_time,dts_time,duration_time,size,flags -of json out.mkv
Traceback (most recent call last):
  File "/home/guru/mytest/.venv/bin/ffmpeg-bitrate-stats", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/guru/mytest/.venv/lib/python3.12/site-packages/ffmpeg_bitrate_stats/__main__.py", line 146, in main
    br.calculate_statistics()
  File "/home/guru/mytest/.venv/lib/python3.12/site-packages/ffmpeg_bitrate_stats/bitrate_stats.py", line 179, in calculate_statistics
    self._calculate_frame_sizes()
  File "/home/guru/mytest/.venv/lib/python3.12/site-packages/ffmpeg_bitrate_stats/bitrate_stats.py", line 269, in _calculate_frame_sizes
    ret[0]["duration"] == "NaN"
    ~~~^^^
IndexError: list index out of range
```

**Environment (please complete the following information):**
- [x] Ubuntu 22.04
- [x] Python 3.12.7
- [x] ffmpeg version n7.1-7-g63f5c007a7-20241010

**Comment by slhck** [comment]
Hm, that's interesting, I can't reproduce this here:

```
➜ python3 -m ffmpeg_bitrate_stats test/out.mkv -v
DEBUG - Calculating frame size from test/out.mkv
DEBUG - [cmd] ffprobe -loglevel error -select_streams v:0 -read_intervals % -show_packets -show_entries packet=pts_time,dts_time,duration_time,size,flags -of json test/out.mkv
DEBUG - Collecting chunks for bitrate calculation
{
    "input_file": "test/out.mkv",
    "stream_type": "video",
    "avg_fps": 30.303,
    "num_frames": 62,
    "avg_bitrate": 2672.293,
    "avg_bitrate_over_chunks": 2614.803,
    "max_bitrate": 2850.544,
    "min_bitrate": 2379.062,
    "max_bitrate_factor": 1.067,
    "bitrate_per_chunk": [
        2850.544,
        2379.062
    ],
    "aggregation": "time",
    "chunk_size": 1.0,
    "duration": 2.046
}
```

This is with:

```
➜ ffprobe -version
ffprobe version 7.1 Copyright (c) 2007-2024 the FFmpeg developers
built with Apple clang version 16.0.0 (clang-1600.0.26.4)
configuration: --prefix=/opt/homebrew/Cellar/ffmpeg/7.1_2 --enable-shared --enable-pthreads --enable-version3 --cc=clang --host-cflags= --host-ldflags='-Wl,-ld_classic' --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libaribb24 --enable-libbluray --enable-libdav1d --enable-libharfbuzz --enable-libjxl --enable-libmp3lame --enable-libopus --enable-librav1e --enable-librist --enable-librubberband --enable-libsnappy --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libsoxr --enable-libzmq --enable-libzimg --disable-libjack --disable-indev=jack --enable-videotoolbox --enable-audiotoolbox --enable-neon
libavutil      59. 39.100 / 59. 39.100
libavcodec     61. 19.100 / 61. 19.100
libavformat    61.  7.100 / 61.  7.100
libavdevice    61.  3.100 / 61.  3.100
libavfilter    10.  4.100 / 10.  4.100
libswscale      8.  3.100 /  8.  3.100
libswresample   5.  3.100 /  5.  3.100
libpostproc    58.  3.100 / 58.  3.100
```

Could you please show me the raw output from the ffprobe command that was printed in your debug log? (Maybe truncated if the output is too long.)

**Comment by ddorian** [comment]
Full output:

```
❯ ffprobe out.mkv
ffprobe version n7.1-7-g63f5c007a7-20241010 Copyright (c) 2007-2024 the FFmpeg developers
  built with gcc 14.2.0 (crosstool-NG 1.26.0.120_4d36f27)
  configuration: --prefix=/ffbuild/prefix --pkg-config-flags=--static --pkg-config=pkg-config --cross-prefix=x86_64-ffbuild-linux-gnu- --arch=x86_64 --target-os=linux --enable-gpl --enable-version3 --disable-debug --enable-iconv --enable-zlib --enable-libfreetype --enable-libfribidi --enable-gmp --enable-libxml2 --enable-openssl --enable-lzma --enable-fontconfig --enable-libharfbuzz --enable-libvorbis --enable-opencl --enable-libpulse --enable-libvmaf --enable-libxcb --enable-xlib --enable-amf --enable-libaom --enable-libaribb24 --enable-avisynth --enable-chromaprint --enable-libdav1d --enable-libdavs2 --enable-libdvdread --enable-libdvdnav --disable-libfdk-aac --enable-ffnvcodec --enable-cuda-llvm --enable-frei0r --enable-libgme --enable-libkvazaar --enable-libaribcaption --enable-libass --enable-libbluray --enable-libjxl --enable-libmp3lame --enable-libopus --enable-librist --enable-libssh --enable-libtheora --enable-libvpx --enable-libwebp --enable-libzmq --enable-lv2 --enable-libvpl --enable-openal --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopenmpt --enable-librav1e --enable-librubberband --disable-schannel --enable-sdl2 --enable-libsoxr --enable-libsrt --enable-libsvtav1 --enable-libtwolame --enable-libuavs3d --enable-libdrm --enable-vaapi --enable-libvidstab --enable-vulkan --enable-libshaderc --enable-libplacebo --enable-libvvenc --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxvid --enable-libzimg --enable-libzvbi --extra-cflags=-DLIBTWOLAME_STATIC --extra-cxxflags= --extra-libs='-ldl -lgomp' --extra-ldflags=-pthread --extra-ldexeflags=-pie --cc=x86_64-ffbuild-linux-gnu-gcc --cxx=x86_64-ffbuild-linux-gnu-g++ --ar=x86_64-ffbuild-linux-gnu-gcc-ar --ranlib=x86_64-ffbuild-linux-gnu-gcc-ranlib --nm=x86_64-ffbuild-linux-gnu-gcc-nm --extra-version=20241010
  libavutil      59. 39.100 / 59. 39.100
  libavcodec     61. 19.100 / 61. 19.100
  libavformat    61.  7.100 / 61.  7.100
  libavdevice    61.  3.100 / 61.  3.100
  libavfilter    10.  4.100 / 10.  4.100
  libswscale      8.  3.100 /  8.  3.100
  libswresample   5.  3.100 /  5.  3.100
  libpostproc    58.  3.100 / 58.  3.100
Input #0, matroska,webm, from 'out.mkv':
  Metadata:
    ENCODER         : Lavf61.7.100
  Duration: 00:00:02.17, start: 0.000000, bitrate: 2681 kb/s
  Stream #0:0: Video: h264 (High), yuv420p(progressive), 1280x720, 30 fps, 30 tbr, 1k tbn (default)
      Metadata:
        DURATION        : 00:00:02.166000000
  Stream #0:1: Audio: aac (LC), 44100 Hz, stereo, fltp (default)
      Metadata:
        title           : simple_aac
        DURATION        : 00:00:02.020000000

```

**Comment by slhck** [comment]
Sorry if I was unclear, I meant the ffprobe command that is printed in the debug log, the one with various options set.

**Comment by ddorian** [comment]
```
ffprobe -loglevel error -select_streams v:0 -read_intervals % -show_packets -show_entries packet=pts_time,dts_time,duration_time,size,flags -of json out.mkv
{
    "packets": [

    ]
}

```

**Comment by slhck** [comment]
I'm afraid I can't help here, as `ffprobe` is returning no information for your file.

Please try with a stable ffmpeg release to see if that fixes the issue; and if so, you can report a bug on https://trac.ffmpeg.org/.

**Comment by slhck** [comment]
PS: I can't reproduce this with `ffprobe version N-117676-g87068b9600-tessus` (from [here](https://evermeet.cx/ffmpeg/)) either.

**Comment by ddorian** [comment]
I use builds from https://github.com/BtbN/FFmpeg-Builds/releases cause it's multi platform. Also tried with yesterdays build https://github.com/BtbN/FFmpeg-Builds/releases/download/autobuild-2024-11-03-13-04/ffmpeg-n7.1-16-g15035aaec0-linux64-gpl-7.1.tar.xz and still didn't work. 

I had this on an automated test so a ffmpeg upgrade broke it but it's a test that I rarely run... I'll figure something out. Thank you!

**Comment by slhck** [comment]
Ok, please do report back if you find out anything useful. I'll implement a warning to more clearly identify cases such as yours as ffprobe-related issues.

**Comment by slhck** [comment]
Could this be dependent on the shell being used? I think it makes sense to not include this option if the user has not specified any intervals.

Please check if v1.1.2 fixes your issue (see 02683ac).

**Comment by ddorian** [comment]
> Could this be dependent on the shell being used?

I don't know, it happened even when quoting "%".

> Please check if v1.1.2 fixes your issue (see https://github.com/slhck/ffmpeg-bitrate-stats/commit/02683accb49750f4342150d66c97ec5675bc6028).

That fixed it.
